clang-tools-extra/clang-tidy/ClangTidy.cpp

```diff
@@ -227,6 +227,14 @@ class ErrorReporter {
           llvm::errs() << "Can't apply replacements for file " << File << "\n";
         }
       }
+
+      auto BuildDir = Context.getCurrentBuildDirectory();
```

Comment:
I think you may need to do what is done in handleErrors method.
Simply remember old value, change new value to build directory, and after call to overwriteChangedFiles restore it.
By default getCurrentWorkingDirectory should return actually current working directory.

Reply:
You're right, the `clang-tidy/infrastructure/clang-tidy-run-with-database.cpp` test kept passing just by chance. I added extra test case to prove that we need to keep closer track for each file what was the CWD when we processed it.